src/main/resources/avro/variants.avdl

```diff
@@ -365,13 +370,18 @@ it is also common for it to be a mix of `Segment`s on reference and novel
 `Sequence`s, or in general to be any contiguous path through the augmented
 sequence graph.
 
-`Allele`s belong to `VariantSet`s. `Allele`s can also be used to represent the
-reference and alternate alleles of `Variant`s.
+`Allele`s are persistent named objects, and may be associated with any number of
+`VariantSet`s and/or `ReferenceSet`s. Like `Sequence`s and `Join`s, `Allele`s
+are inherited when one `ReferenceSet` includes another. Zero or more `Variant`s
+in a `VariantSet` may use a single `Allele`.
 
-Note that `Path`s cannot follow `Join`s not represented in the augmented
-sequence graph. For example, if  an `Allele` that spans a novel deletion is
-required, a new `Join` should exist to describe that deletion, available through
-`searchJoins()`.
+To get the `Allele`s in a `ReferenceSet` or `VariantSet`, use `searchAlleles()`.
+
+Note that `Path`s cannot follow `Join`s not represented in the relevant sequence
+graph. For example, if  an `Allele` in a `VariantSet` describes a novel deletion
+relative to the applicable `ReferenceSet`, the `Join` describing that deletion
+must be a novel `Join` in that `VariantSet`, and thus available through
+`searchJoins()` on the `VariantSet`.
```

Comment:
Yes, and that allele could be in the ReferenceSet or in the VariantSet.

Maybe I should add the proviso that it can't be in a VariantSet if it's in that VariantSet's ReferenceSet...